test/test_abstractdata.jl

```diff
@@ -412,9 +412,9 @@ x2 = Ket(b_r, rand(ComplexF64, length(b_r)))
 xbra1 = Bra(b_l, rand(ComplexF64, length(b_l)))
 xbra2 = Bra(b_l, rand(ComplexF64, length(b_l)))
 
-# Addition
-@test_throws ArgumentError op1 + op2
-@test_throws ArgumentError op1 - op2
+# Addition Addition of LazyTensor now returns LazySum
```

Comment:
Before merge I would suggest deleting the commented-out tests. They would not be informative to future readers, long after this merge. Future readers should use `git blame` if they want to know why a change was made.

Could you add tests for the failure modes of this contribution? E.g., `@test_throws` for where your code specifically raises an error to warn about something it can not do. And are there any other implicit assumptions about shapes and bases and other things? We do not want the user to obtain nonsense results if they do not happen to know that some combination of operations is not supported.

Reply:
Updated code to no longer assume anything about BL == BR and added test to ensure that errors are thrown if addition between operators is not allowed.